content/accessible/labels-and-text-alternatives/index.md

```diff
@@ -0,0 +1,334 @@
+      </td>
+      <td>
+        <a href="#text-alternatives-for-images-and-objects">
+          Text alternatives for images and objects
+        </a>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        Buttons have an accessible name
+      </td>
+      <td>
+        <a href="#label-buttons-and-links">Label buttons and links</a>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        Links have a discernible name
+      </td>
+      <td>
+        <a href="#label-buttons-and-links">Label buttons and links</a>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        Form elements have associated labels
+      </td>
+      <td>
+        <a href="#label-form-elements">Label form elements</a>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Label documents and frames
+Every page should have a
+[`title`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/title)
+element that briefly explains what the page is about. The `title` element gives
+the page its accessible name. When a screen reader enters the page, this is the
+first text that will be announced.
+
+<pre class="prettyprint devsite-disable-click-to-copy">
+&lt;!doctype html&gt;
+  &lt;html lang=&quot;en&quot;&gt;
+    &lt;head&gt;
+      <strong>&lt;title&gt;Mary's Maple Bar Fast-Baking Recipe&lt;/title&gt;</strong>
+    &lt;/head&gt;
+  &lt;body&gt;
+    …
+  &lt;/body&gt;
+&lt;/html&gt;
+</pre>
+
+<div class="aside note">
+For tips on writing effective titles, see the
+<a href="https://web.dev/discoverable/write-descriptive-text">
+Write descriptive titles guide
+</a>.
+</div>
+
+Similarly, any `frame` or `iframe` elements should have a `title` attribute.
+
+```html
+<iframe title="An interactive map of San Francisco" src="…"></iframe>
+```
+
+While an `iframe` may contain its own internal `title` element, a screen
+reader will usually stop at the frame boundary and announce the element's
+role—"frame"—and its accessible name, provided by the `title` attribute. This
+lets the user to decide if they wish to enter the frame or bypass it.
+
+## Text alternatives for images and objects
+An `img` should always be accompanied by an
+[`alt`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#Attributes)
+attribute, to give the image its accessible name. If the image fails to load,
+the `alt` text will be used as a placeholder so users have a sense of what the
+image was trying to convey.
+
+Writing good `alt` text is a bit of an art, but there are a couple of guidelines
+you can follow:
+
+1. Determine if the image provides content that would otherwise be difficult to
+attain from reading the surrounding text.
+1. If so, convey the content as succinctly as possible.
+
+If the image is more of a decoration, and does not provide any useful content,
+you can give it an empty `alt=""` attribute to remove it from the accessibility
+tree.
+
+<div class="aside note">
+You can learn more about writing effective <code>alt</code> by checking out <a
```

Comment:
Just to avoid multiple 'You can' phrases, maybe just make inside the note.... 'Learn more about'